archaius-core/src/main/java/com/netflix/config/DynamicBooleanProperty.java

```diff
@@ -23,14 +23,41 @@
  *
  */
 public class DynamicBooleanProperty extends PropertyWrapper<Boolean> {
+
+    protected volatile boolean primitiveValue;
+
     public DynamicBooleanProperty(String propName, boolean defaultValue) {
         super(propName, Boolean.valueOf(defaultValue));
+
+        // Set the initial value of the cached primitive value.
+        this.primitiveValue = chooseValue();
+
+        // Add a callback to update the cached primitive value when the property is changed.
+        this.prop.addCallback(new Runnable() {
+            @Override
+            public void run() {
+                primitiveValue = chooseValue();
+            }
+        });
     }
+
     /**
      * Get the current value from the underlying DynamicProperty
+     *
+     * @return
+     */
+    private boolean chooseValue() {
+        Boolean propValue = this.prop == null ? null : this.prop.getBoolean();
```

Comment:
Actually, all of the overhead does seem to be caused by this. I've added another benchmark comparison to the CachedPropertiesPerfTest to illustrate. eg:

    Original DynamiceBooleanProperty totalled 7087 ms.
    New DynamicBooleanProperty totalled 1252 ms.
    New DynamicBooleanProperty with old get() totalled 7129 ms.